modules/oci-ocne-autoscaling/variables.tf

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2023 Oracle Corporation and/or affiliates.  All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+// Common OCI resource variables
+variable "availability_domain_id" {
+  type        = string
+  description = "The ID of the availability domain inside the `region` in which to create this deployment"
+}
+
+variable "compartment_id" {
+  type        = string
+  description = "The OCID of the compartment in which to create this deployment"
+}
+
+variable "prefix" {
+  type        = string
+  description = "A unique prefix to attach to the name of all cloud resources that are created as a part of this deployment"
+}
+
+variable "instance_shape" {
+  type        = string
+  description = "The OCI instance shape to use for all compute resources that are created as part of this deployment"
+}
+
+variable "image_ocid" {
+  type        = string
+  description = "The OCID of the OS image to use when creating all compute resources that are part of this deployment"
+}
+
+variable "subnet_id" {
+  type        = string
+  description = "The OCID of a pre-existing subnet that all newly created cloud resources will be configured to use.  If this variable to set to the empty string, a network configuration will be generated automatically"
+}
+
+// Compute instance specific variables
+variable "ssh_public_key_path" {
+  type        = string
+  description = "The SSH public key path to use when configuring access to any compute resources created as part of this deployment"
+}
+variable "instance_user_data" {
```

Comment:
I think this should:
1. have multiple: control plane, workers so you can have different configuration
2. renamed accordingly